scrc/dataset_creation/criticality_dataset_creator.py

```diff
@@ -1,93 +1,359 @@
+        except ValueError as ve:
+            self.logger.info("citations could not be extracted to dict")
+            self.logger.info(citations_as_string)
+        if cits:  # only return something if we actually have citations
+            return cits
+
+    def get_file_number(self, citation):
+        """
+        find for each citation string the matching citation from the start of bge (first page)
+        :param citation:         citation as string as found in text
+        :return:                 RulingCitation always in German
+        """
+        # handle citation always in German
+        found_citation = RulingCitation(citation, 'de')
+        if str(found_citation) in self.available_bges:
+            return found_citation
+        else:
+            # find closest bge with smaller page_number
+            year = found_citation.year
+            volume = found_citation.volume
+            page_number = found_citation.page_number
+            new_page_number = -1
+            for match in self.available_bges:
+                if f"BGE {year} {volume}" in match:
+                    tmp = RulingCitation(match, 'de')
+                    if new_page_number < tmp.page_number <= page_number:
+                        new_page_number = tmp.page_number
+            return RulingCitation(f"{year} {volume} {new_page_number}", 'de')
+
+    def save_huggingface_dataset(self, splits, feature_col_folder):
+        """
+        save data as huggingface dataset with columns: 'id', 'year': year, 'language',
+        region', 'canton', 'legal area', 'bge_label', 'considerations' and 'facts'
+        :param splits:                  specifying splits of dataset
+        :param feature_col_folder:      name of folder
+        """
+        huggingface_dir = self.create_dir(feature_col_folder, 'huggingface')
+        for split in ['train', 'val', 'test']:
+            records = []
+            df = splits[split]
+
+            tuple_iterator = zip(df.index, df['year'], df['legal_area'], df['origin_region'],
+                                 df['origin_canton'], df['bge_label'], df['lang'], df['considerations'], df['facts'])
+
+            for case_id, year, legal_area, region, canton, bge_label, lang, consideration, fact in tuple_iterator:
+                if not isinstance(canton, str) and (canton is None or math.isnan(canton)):
+                    canton = 'n/a'
+                if not isinstance(region, str) and (region is None or math.isnan(region)):
+                    region = 'n/a'
+                if not isinstance(legal_area, str) and (legal_area is None or math.isnan(legal_area)):
+                    legal_area = 'n/a'
+                record = {
+                    'id': case_id,
+                    'year': year,
+                    'language': lang,
+                    'region': ' '.join(region.split('_')),
+                    'canton': canton,
+                    'legal area': ' '.join(legal_area.split('_')),
```

Comment:
please always use underscores. Whitespace could lead to problems when trying to access it.